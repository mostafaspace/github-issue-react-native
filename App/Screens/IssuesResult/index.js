import React from 'react';
import {View, Text, FlatList, TouchableOpacity} from 'react-native';
import {Indicator} from '../../Components/Indicator';
import {IssueCard} from '../../Components/IssueCard';
import {NoRecord} from '../../Components/NoRecord';
import Icon from 'react-native-vector-icons/MaterialIcons';
import {Issues} from '../../Controllers';
import {Actions} from 'react-native-router-flux';
import {useSelector, useDispatch} from 'react-redux';
import {APPEND_TO_LIST, LAST_PAGE, LOAD_MORE} from '../../Actions/IssuesActions';
import Styles from './styles';

export default function IssuesResult() {
    const {IssuesList, IsLoading, IsLoadingMore, Organization, Repo, Page, LastPage} = useSelector((state) => state);
    const dispatch = useDispatch();

    function loadMore() {
        if (!LastPage) {
            dispatch({type: LOAD_MORE});
            Issues({Organization, Repo, Page}).then((newList) => {
                if (newList && newList.length > 0) {
                    dispatch({type: APPEND_TO_LIST, payload: newList});
                } else {
                    dispatch({type: LAST_PAGE});
                }
            });
        }
    }

    return (
        <View style={Styles.container}>
            {IsLoading ? (
                <Indicator />
            ) : IssuesList && IssuesList.length > 0 ? (
                <FlatList
                    data={IssuesList}
                    renderItem={({item, index}) => <IssueCard key={'issue' + index} issue={item} />}
                    contentContainerStyle={Styles.flatContainerStyle}
                    keyExtractor={(item, index) => index.toString()}
                    onEndReached={loadMore}
                    onEndReachedThreshold={0.2}
                    ListFooterComponent={() =>
                        IsLoadingMore ? (
                            <View style={Styles.moreLoad}>
                                <Indicator size={'small'} />
                            </View>
                        ) : (
                            <View />
                        )
                    }
                />
            ) : (
                <NoRecord text={'No Issues Found'} />
            )}
        </View>
    );
}

IssuesResult.navigationOptions = () => {
    return {
        headerLeft: () => (
            <TouchableOpacity onPress={() => Actions.pop()}>
                <Icon name={'keyboard-backspace'} style={Styles.backIcon} />
            </TouchableOpacity>
        ),
        headerTitle: () => <Text style={Styles.headerTitle}>{'Issues List'}</Text>,
    };
};
