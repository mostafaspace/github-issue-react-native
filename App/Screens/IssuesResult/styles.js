import {StyleSheet} from 'react-native';

export default StyleSheet.create({
    container: {
        flex: 1,
    },
    headerTitle: {
        fontWeight: 'bold',
        fontSize: 16,
    },
    flatContainerStyle: {
        paddingVertical: 16,
        paddingHorizontal: 8,
    },
    moreLoad: {
        marginBottom: 16,
    },
    backIcon: {
        width: 24,
        height: 24,
        fontSize: 24,
        color: 'black',
        marginHorizontal: 16,
    },
});
