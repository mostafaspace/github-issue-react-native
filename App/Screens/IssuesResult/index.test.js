import React from 'react';
import {render, cleanup} from '@testing-library/react-native';
import IssuesResult from '../IssuesResult';
import renderer from 'react-test-renderer';
import '@testing-library/jest-native/extend-expect';

afterEach(cleanup);
it('renders component', () => {
    render(<IssuesResult />);
});

it('IssuesResult matches snapshot', () => {
    const tree = renderer.create(<IssuesResult />).toJSON();
    expect(tree).toMatchSnapshot();
});
