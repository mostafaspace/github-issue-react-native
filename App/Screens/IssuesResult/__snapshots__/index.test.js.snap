// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssuesResult matches snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
    testID="NoRecord"
  >
    <Text
      style={
        Object {
          "color": "#222",
          "lineHeight": 18,
          "margin": 16,
          "textAlign": "center",
        }
      }
    >
      No Issues Found
    </Text>
  </View>
</View>
`;
